include/dyno/concept.hpp

```diff
@@ -63,6 +63,21 @@ namespace detail {
       return boost::hana::make_basic_tuple(boost::hana::first(c)...);
     }
   };
+
+  template <typename... Clauses>
+  struct check_duplicate_names {
+    template <typename... Pairs>
+    struct check_duplicate_names_impl{};
+  
+    template <typename... Pairs>
+    struct check_duplicate_names_impl<const boost::hana::basic_tuple<Pairs...>> {
+      static constexpr bool value = boost::hana::detail::has_duplicates<
```

Comment:
You know you shouldn't be using stuff from a library's `detail::` namespace, right? :-)